dojo/tools/gitlab_secret_detection_report/parser.py

```diff
@@ -0,0 +1,95 @@
+import hashlib
+import json
+from dojo.models import Finding
+
+
+class GitlabSecretDetectionReportParser(object):
+    """
+    GitLab's secret detection report
+    See more: https://gitlab.com/gitlab-org/security-products/security-report-schemas/-/blob/master/dist/secret-detection-report-format.json
+    """
+
+    def get_scan_types(self):
+        return ["GitLab Secret Detection Report"]
+
+    def get_label_for_scan_types(self, scan_type):
+        return "GitLab Secret Detection Report"
+
+    def get_description_for_scan_types(self, scan_type):
+        return "GitLab Secret Detection Report file can be imported in JSON format (option --json)."
+
+    def get_findings(self, file, test):
+        # Load JSON data from uploaded file
+        data = json.load(file)
+
+        # Initial dataset to be deduplicate via JSON key
+        dupes = dict()
+
+        # This is required by schema - it won't be null / undefined
+        date = data["scan"]["end_time"]
+
+        # Vulnerabilities is stored on vulnerabilities key
+        vulnerabilities = data["vulnerabilities"]
+        for vulnerability in vulnerabilities:
+            title = vulnerability["message"]
+            description = vulnerability["description"]
+            severity = self.normalise_severity(vulnerability["severity"])
+
+            file_path = ""
+            try:
+                file_path = vulnerability["location"]["file"]
+            except:
+                pass
+
+            line = -1
+            try:
+                line = int(vulnerability["location"]["start_line"])
+            except:
+                pass
+
+            raw_source_code = ""
+            try:
+                raw_source_code = vulnerability["raw_source_code_extract"]
+            except:
+                pass
+
+            finding = Finding(
+                test=test,
+                title=title,
+                description=description,
+                date=date,
+                severity=severity,
+                static_finding=True,
+                dynamic_finding=False,
+            )
```

Comment:
it seems that you have an id in the file. Could you set `unique_id_from_tool` ?

Reply:
added